components/ui/data-table/use-data-table-styles.tsx

```diff
@@ -0,0 +1,56 @@
+import { DataTableColumnDef } from "./data-table";
+
+interface useDataStylesInput<TData, TValue> {
+  columns: DataTableColumnDef<TData, TValue>[];
+}
+
+interface GenerateClassnameForHeaderInput {
+  columnId: string;
+}
+
+interface GenerateClassnameForCellInput {
+  columnId: string;
+}
+
+export const useDataTableStyles = <TData, TValue>(input: useDataStylesInput<TData, TValue>) => {
+  const { columns } = input;
+
+  const generateClassnameForHeader = (input: GenerateClassnameForHeaderInput) => {
+    const { columnId } = input;
+
+    const baseClassnameForHeader = "border-b bg-white";
+    const classnameForLeftColumnFixation = "sticky left-0 border-r";
+
+    const columnDefinition = columns.find((column) => column.accessorKey === columnId);
+
+    if (!columnDefinition || columnDefinition.fixed !== true) {
+      return baseClassnameForHeader;
+    }
+
+    return baseClassnameForHeader
+      .concat(" ")
+      .concat(classnameForLeftColumnFixation);
+  };
+
+  const generateClassnameForCell = (input: GenerateClassnameForCellInput) => {
+    const { columnId } = input;
+
+    const baseClassnameForHeader = "border-b bg-white group-hover:bg-zinc-100";
```

Comment:
I made the hover color a little bit more opaque. Honestly I think this looks fine too. If you're wondering why I couldn't keep the `bg-zinc:100/50`, the sticky column being partially transparent lets you see through it to the columns underneath and it looked really bad. Trying to get some kind of background white div behind it was painful too.

Honestly, I think this looks totally fine. If the color is to harsh I'm sure we can do `zinc-50` or something like that. Just bringing attention to it in case someone has a strong opinion here but I think we're all good honestly and I just can't imagine that preserving the original color is worth the headache.

Pics below are before and after.

![Peek 2024-01-06 14-22](https://github.com/PathoTracker/Pathotracker/assets/86806388/16f494de-c496-4c94-b710-8392db32e027)

![Peek 2024-01-06 14-23](https://github.com/PathoTracker/Pathotracker/assets/86806388/8be74276-e0be-4815-87ac-9f9792d5f4d3)